server/controllers/finance/vouchers.js

```diff
@@ -0,0 +1,152 @@
+/**
+* The /vouchers/ HTTP API endpoint
+*
+* @module finance/vouchers
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to voiuchers transactions and define all vouchers api functions
+*
+* @required node-uuid
+* @required lib/db
+*/
+
+'use strict';
+
+var _    = require('lodash'),
+    uuid = require('node-uuid'),
+    db   = require('../../lib/db');
+
+/** Get list of vouchers */
+exports.list = getList;
+
+/** Get detail of vouchers */
+exports.detail = getDetail;
+
+/** Create a new voucher record */
+exports.create = create;
+
+/**
+* GET /vouchers
+*
+* @method getList
+*/
+function getList(req, res, next) {
+  var query =
+    'SELECT v.uuid, v.date, v.project_id, v.reference, v.currency_id, v.amount, ' +
+    'v.description, v.document_uuid, v.user_id, vi.uuid AS voucher_item_uuid, ' +
+    'vi.account_id, vi.debit, vi.credit ' +
+    'FROM voucher v JOIN voucher_item vi ON vi.voucher_uuid = v.uuid ';
+
+  var builder = queryCondition(query, req.query);
+
+  db.exec(builder.query, builder.conditions)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* GET /vouchers/:uuid
+*
+* @method getDetail
+*/
+function getDetail(req, res, next) {
+  var query =
+    'SELECT v.uuid, v.date, v.project_id, v.reference, v.currency_id, v.amount, ' +
+    'v.description, v.document_uuid, v.user_id, vi.uuid AS voucher_item_uuid, ' +
+    'vi.account_id, vi.debit, vi.credit ' +
+    'FROM voucher v JOIN voucher_item vi ON vi.voucher_uuid = v.uuid ' +
+    'WHERE v.uuid = ? ;';
+
+  db.exec(query, req.params.uuid)
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* POST /vouchers
+*
+* @method create
+*/
+function create(req, res, next) {
+
+  if (!checkForMissing(req.body)) { return next(new req.codes.ERR_MISSING_REQUIRED_PARAMETERS()); }
```

Comment:
This would be more readable if you used multiple new lines.  Like this:
```js
if (!checkForMissing(req.body)) {
  return next(new req.codes.ERR_MISSING_REQUIRED_PARAMETERS()); 
}
```
Also, `checkForMissing()` isn't a very descriptive name.  A name should ideally describe the required parameter and the return type.  In this case, I would name this function `hasMissingProperties()`.  That way, we know the following information:
 1.  "Properties" means we are checking an object's properties -- the param is an object, not an array, function, or any other type.
 2. "hasMissing" means that this function will answer the question with a "yes" or "no" -- a boolean type.  We expect it to return `true` or `false`.